File: 250424/비오는 날/rainy-day.cpp
#include <iostream>
#include <string>
using namespace std;

class Forecast {
    public:
        string date;
        string day;
        string weather;
        Forecast(string date, string day, string weather) {
            this->date = date;
            this->day = day;
            this->weather = weather;
        }
        Forecast() {};
};

Forecast ans = Forecast("9999-99-99", "", "");

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    string date, day, weather;
    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> date >> day >> weather;
        Forecast forecast = Forecast(date, day, weather);
        if(weather == "Rain") {
            if(ans.date >= forecast.date) {
                ans = forecast;
            }
        }
    }
    cout << ans.date << " " << ans.day << " " << ans.weather;
    return 0;
}

File: 250424/비오는 날/rainy-day.java
import java.util.Scanner;
import java.util.Arrays;

class Program {
    String date;
    String day;
    String weather;

    public Program() {};
    public Program(String date, String day, String weather) {
        this.date = date;
        this.day = day;
        this.weather = weather;
    }
}
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int index = 0;
        int min_year = Integer.MAX_VALUE;
        int min_month = Integer.MAX_VALUE;
        int min_day = Integer.MAX_VALUE;
        Program[] program = new Program[n];
        for (int i = 0; i < n; i++) {
            String date = sc.next();
            String day = sc.next();
            String weather = sc.next();
            program[i] = new Program(date, day, weather);
        }
        for(int i = 0; i < n; i++) {
            if(program[i].weather.equals("Rain")) {
                String[] date = program[i].date.split("-");
                int year = Integer.parseInt(date[0]);
                int month = Integer.parseInt(date[1]);
                int day = Integer.parseInt(date[2]);
                if(year < min_year) {
                    min_year = year;
                    min_month = month;
                    min_day = day;
                    index = i;
                } else if(year == min_year) {
                    if(month < min_month) {
                        min_month = month;
                        min_day = day;
                        index = i;
                    } else if(month == min_month) {
                        if(day < min_day) {
                            min_day = day;
                            index = i;
                        }
                    }
                }
            }
        }
        System.out.println(program[index].date + " " + program[index].day + " " + program[index].weather);
        sc.close();
    }
}